File: programmers/난이도별/level01.최대공약수와_최소공배수/areum.js
function solution(n, m) {
    var answer = [];
    var temp = 0;
    
     // n자리에 무조건 작은 수가 오게하기
    if(n>m){
        temp = m;
        m = n;
        n = temp;
    }
    
    // 최대공약수
    for(var i=n; i>=1; i--){
        if(n%i===0 && m%i===0){
            answer[0] = i;
            break;
        }
    }
    
    // 짠 코드인데 다른 사람의 코드가 더 효율적임
    // for(let i=1; i<=m; i++){
    //     if(n%i==0 && m%i==0){
    //         answer[0]=i;
    //     }
    // }
    
    // 최소공배수
    answer[1] = m*n/answer[0];
    
    return answer;
}

File: programmers/난이도별/level01.소수_찾기/areum.java
class Solution {
      public int solution(int n) {
          int answer = 0;
          
          int[] number = new int[n+1];
          
          //2부터 n까지의 수를 배열에 넣는다.
          for(int i=2; i<=n; i++) {
              number[i] = i;
          }
          
          //2부터 시작해서 그의 배수들을 0으로 만든다.
          //후에 0이면 넘어가고 아니면 그의 배수들을 다시 0으로 만든다.
          for(int i=2; i<=n; i++) {
              if(number[i]==0) continue;
              
              for(int j= 2*i; j<=n; j += i) {
                  number[j] = 0;
              }
          }
          
          //배열에서 0이 아닌 것들의 개수를 세준다.
          for(int i=0; i<number.length; i++) {
              if(number[i]!=0) {
                  answer++;
              }
          }
          
          return answer;
      }
    }
